app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,19 @@
+class VotesController < ApplicationController
+  def create
+    if !session[:user_id]
+      flash[:message] = "A problem occured: You must log in to do that"
+      redirect_back(fallback_location: root_path)
+      return
+    elsif Vote.find_by(user_id: session[:user_id], work_id: params[:work_id])
+      flash[:message] = "You have already voted for this work"
+      redirect_back(fallback_location: root_path)
+      return
+    elsif session[:user_id] 
```

Comment:
This doesn't need to be an `elsif`, as you know after the initial `if` that we at least have a user.